### ISSUE TITLE
Type.GetInterfaceMap with generic interface types

### ISSUE THREAD

**Issue by pavel-agarkov** [dotnet/docs#6796]
When type parameter of `Type.GetInterfaceMap` method is represented by:
1) Open generic constructed type - method throws `ArgumentException` `"interface not found"` which is not mentioned in the docs
2) Closed generic constructed type - method returns `InterfaceMapping` with `InterfaceType` property equal to the specified `type` parameter, but not the actual version of implemented generic interface. Generic parameters in the returned type correspond to the specified `type` parameter, but not to the actual implemented interface type.

> dotnet core 2.1

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f037e068-f720-1ece-533e-b71e5de9b393
* Version Independent ID: 9e8ef13d-1ff6-b78e-494a-57227e2225a4
* Content: [Type.GetInterfaceMap(Type) Method (System)](https://docs.microsoft.com/en-us/dotnet/api/system.type.getinterfacemap?view=netcore-2.1#System_Type_GetInterfaceMap_System_Type_)
* Content Source: [xml/System/Type.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System/Type.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by rpetrusha** [comment]
@pavel-agarkov, I've opened dotnet/dotnet-api-docs#622 to add the exception information. However, I can't reproduce your statement that 
> If the current Type represents a constructed generic type, type parameters are replaced by the appropriate type arguments in the elements of the InterfaceMapping returned by this method. 

Do you have any code that illustrates this?

**Comment by pavel-agarkov** [comment]
@rpetrusha Exactly! This statement is from the docs - not mine. And it is incorrect.
1) The following code will throw an error: `System.ArgumentException: 'Interface not found.'`
```cs
typeof(List<int>).GetInterfaceMap(typeof(IList<>));
```
2) The following code will print ```Program+IHandler`1[System.Exception]``` 
instead of ```Program+IHandler`1[System.ArgumentException]```
```cs
internal class Program
{
    interface IHandler<out T> where T : Exception
    {
    }

    class ArgumentHandler : IHandler<ArgumentException>
    {
    }

    static void Main(string[] args)
    {
        Console.WriteLine(
            typeof(ArgumentHandler)
            .GetInterfaceMap(typeof(IHandler<Exception>))
            .InterfaceType);
        Console.ReadKey();
    }
}
```
So type parameters are replaced by whatever were specified in the `interfaceType` and not 'appropriate type' as mentioned in the docs. This behavior made the method useless for my case.
